File: Ruby/golem.rb
class Golem
  attr_accessor :hp
  attr_accessor :defense
  attr_accessor :attack

  def initialize(hp = 300, defense = 80, attack = 50)
    @hp = hp
    @defense = defense
    @attack = attack
  end
end

class GolemEx
  GOLEM_HIT = 70
  GOLEM_FLEE = 20
  attr_accessor :hp
  attr_accessor :defense
  attr_accessor :attack
  attr_reader :hit
  attr_reader :flee

  def initialize(hp, defense, attack)
    @hp = hp
    @defense = defense
    @attack = attack
    @hit = GOLEM_HIT
    @flee = GOLEM_FLEE
  end
end

File: Kotlin/src/Golem.kt
const val GOLEM_HIT = 70
const val GOLEM_FLEE = 20

class Golem(__hp: Int = 300, __defense: Int = 80, __attack: Int = 50) {
    var hp: Int = __hp
    var defense: Int = __defense
    var attack: Int = __attack
}

data class GolemEx(var Hp: Int, val Defense: Int, val Attack: Int, val Hit: Int = GOLEM_HIT, val Flee: Int = GOLEM_FLEE)